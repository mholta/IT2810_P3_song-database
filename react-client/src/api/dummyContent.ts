import { Album, Artist, Song } from './types';
export const dummyArtist: Artist = {
  id: 'acta',
  name: 'Acta',
  location: 'Stavanger',
  webPage: 'actabibelskole.no',
  picture:
    'https://uploads-ssl.webflow.com/5ed95215f8138049de460b43/605cf392565470abf2a035d4_acta_lovsang_2020.jpg',
};

export const dummyAlbum: Album = {
  id: 'motepunkt-acta-lovsang',
  title: 'Møtepunkt',
  artists: [dummyArtist],
  picture:
    'https://uploads-ssl.webflow.com/5ed95215f8138049de460b43/604374cdc48ec430aa6d4894_M%C3%B8tepunkt%20-%20cover.jpeg',
  releaseDate: new Date(
    'Fri Jul 10 2020 00:00:00 GMT+0000 (Coordinated Universal Time)'
  ),
  publisher: 'IMI-kirken',
  producers: ['Tore Kulleseid'],
  iTunes: 'https://music.apple.com/no/album/møtepunkt-single/1520693028',
  spotify: 'https://open.spotify.com/album/6WzBmm9l5zp1HnfKRVCAKx',
};

export const dummySong: Song = {
  id: 'all-pris-acta-lovsang',
  title: 'All pris',
  album: dummyAlbum,
  tempo: '68',
  key: 'F#',
  time: '4/4',
  artists: [dummyArtist],
  writers: ['Eline Walderhaug', 'Thomas Wilhelmsen', 'Tore Kulleseid'],
  contributors: [
    'Tobias Valdal Høssung (trommer)',
    'Ruben Strømme (bass)',
    'Anders Kjøllesdal Hansen (keys)',
    'Vemund Aasemoen Aardal (keys)',
    'Samuel Hilleren (keys)',
    'Aleksander Øvergaard (gitar)',
    'Mads Even Vold (gitar)',
    'Mathilde Eriksen (vokal)',
    'Eline Walderhaug (kor)',
    'Vemund Aasemoen Aardal (kor)',
  ],
  producers: ['Tore Kulleseid'],
  releaseDate: new Date(
    'Fri Jul 10 2020 00:00:00 GMT+0000 (Coordinated Universal Time)'
  ),
  iTunes: 'https://music.apple.com/no/album/møtepunkt-single/1520693028',
  spotify: 'https://open.spotify.com/track/7kXJtTxucwufmP2f32yeJo',
};
