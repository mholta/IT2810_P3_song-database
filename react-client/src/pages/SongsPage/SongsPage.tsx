import React from 'react';
import { styled } from '@mui/system';
import SearchOptions from '../../components/SearchFilter/SearchFilter';
import { gql } from '@apollo/client';
import {
  FilterOptions,
  SortOptions,
  useFilterParams,
  useSortParams,
} from '../../hooks/useSearchParams';
import SearchResults from '../../components/SearchResults/SearchResults';

const SearchResultsPage = () => {
  const filterParams: FilterOptions = useFilterParams();
  const sortParams: SortOptions = useSortParams();

  return (
    <SearchResultsPageWrapper>
      <SearchOptions />
      <SearchResults
        query={GET_SEARCH_RESULTS}
        limit={20}
        options={{
          variables: {
            ...filterParams,
            ...sortParams,
          },
        }}
      />
    </SearchResultsPageWrapper>
  );
};

const SearchResultsPageWrapper = styled('div')`
  padding-top: 4rem;
`;

export const GET_SEARCH_RESULTS = gql`
  query Songs(
    $searchString: String
    $themes: [String!]
    $sortOrder: SortOrder!
    $sortType: SortType!
    $page: Int!
    $limit: Int!
  ) {
    songs(
      searchString: $searchString
      filter: { categories: $themes }
      limit: $limit
      page: $page
      sorting: { order: $sortOrder, sortType: $sortType }
    ) {
      songs {
        _id
        artists {
          name
        }
        title
        album {
          picture
        }
      }
      pages
    }
  }
`;

export default SearchResultsPage;
