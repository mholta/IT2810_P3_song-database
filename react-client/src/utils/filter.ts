/**
 * Compares two booleans for sorting.
 */
export const compareBoolean = (a: boolean, b: boolean) => {
  if (a && !b) return -1;
  if (!a && b) return 1;
  return 0;
};
