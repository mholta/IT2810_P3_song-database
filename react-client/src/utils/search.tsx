export const getQueryStringFromFormSubmitEvent = (
  formData: FormData
): string => {
  return new URLSearchParams(formData as URLSearchParams).toString();
};
