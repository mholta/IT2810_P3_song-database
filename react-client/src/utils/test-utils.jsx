// Copied and modified from https://redux.js.org/usage/writing-tests

import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from '../store/index';
import PageLayoutWrapper from '../components/layout/PageLayoutWrapper';
import { ThemeProvider } from '@mui/material';
import { theme } from '../styles/theme';
import { MemoryRouter, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';

function render(
  ui,
  gqlMocks = {},
  url = '/',
  params = {},
  {
    preloadedState,
    store = createStore(rootReducer, composeWithDevTools()),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    let initEntry = url;
    for (let key of Object.keys(params)) {
      initEntry = initEntry.replace(':' + key, params[key]);
    }

    return (
      <MockedProvider mocks={gqlMocks}>
        <Provider store={store}>
          <ThemeProvider theme={theme}>
            <MemoryRouter initialEntries={[initEntry]}>
              <PageLayoutWrapper>
                <Route path={url}>{children}</Route>
              </PageLayoutWrapper>
            </MemoryRouter>
          </ThemeProvider>
        </Provider>
      </MockedProvider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };
