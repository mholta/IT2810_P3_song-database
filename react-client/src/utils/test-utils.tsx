import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from '../store/index';
import PageLayoutWrapper from '../components/layout/PageLayoutWrapper';
import { ThemeProvider } from '@mui/material';
import { theme } from '../styles/theme';
import { MemoryRouter, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';

interface TestProps {
  children: any;
  gqlMocks?: any;
  url: string;
  params?: any;
}

export const TestProvider = ({
  children,
  gqlMocks = [],
  url,
  params = {},
}: TestProps) => {
  let initEntry = url;
  for (let key of Object.keys(params)) {
    initEntry = initEntry.replace(':' + key, params[key]);
  }

  const store = createStore(rootReducer, composeWithDevTools());

  return (
    <MockedProvider mocks={gqlMocks}>
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <MemoryRouter initialEntries={[initEntry]}>
            <PageLayoutWrapper>
              <Route path={url}>{children}</Route>
            </PageLayoutWrapper>
          </MemoryRouter>
        </ThemeProvider>
      </Provider>
    </MockedProvider>
  );
};
