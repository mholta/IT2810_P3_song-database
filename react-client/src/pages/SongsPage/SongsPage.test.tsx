import { TestProvider } from '../../utils/test-utils';
import renderer from 'react-test-renderer';
import SongsPage, { GET_SEARCH_RESULTS } from './SongsPage';
import { dummySong, dummySongList } from '../../api/testData/dummyContent';

const mockData = [
  {
    request: {
      query: GET_SEARCH_RESULTS,
      variables: { searchString: null, contributor: null, themes: [] },
    },
    result: {
      data: { songs: { songs: dummySongList } },
    },
  },
  {
    request: {
      query: GET_SEARCH_RESULTS,
      variables: { searchString: 'all pris', contributor: null, themes: [] },
    },
    result: {
      data: { songs: { songs: [dummySong] } },
    },
  },
  {
    request: {
      query: GET_SEARCH_RESULTS,
      variables: {
        searchString: 'songnotfound',
        contributor: null,
        themes: [],
      },
    },
    result: {
      data: { songs: { songs: [] } },
    },
  },
];

describe('SongsPage', () => {
  test('rendered successfully', () => {
    const tree = renderer
      .create(
        <TestProvider gqlMocks={mockData} url="/search">
          <SongsPage />
        </TestProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('rendered successfully with query', () => {
    const tree = renderer
      .create(
        <TestProvider gqlMocks={mockData} url="/search?query=all+pris">
          <SongsPage />
        </TestProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('rendered successfully without results', () => {
    const tree = renderer
      .create(
        <TestProvider gqlMocks={mockData} url="/search?query=songnotfound">
          <SongsPage />
        </TestProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
