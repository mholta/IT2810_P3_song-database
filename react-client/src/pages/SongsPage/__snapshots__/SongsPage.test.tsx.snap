// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered successfully 1`] = `
<div
  className="css-185wnrw"
  style={
    Object {
      "height": "768px",
    }
  }
>
  <nav
    className="css-1ebp07k"
    open={false}
  >
    <div
      className="css-ydsbw5"
    >
      <a
        className="css-1yg7sek"
        href="/"
        onClick={[Function]}
      >
         
        <span>
          Hjem
        </span>
      </a>
      <a
        className="css-1yg7sek"
        href="/search"
        onClick={[Function]}
      >
         
        <span>
          Sanger
        </span>
      </a>
      <a
        className="css-1yg7sek"
        href="/search?query=søker på dette"
        onClick={[Function]}
      >
         
        <span>
          Søk på noe
        </span>
      </a>
      <a
        className="css-1yg7sek"
        href="/song/testsang"
        onClick={[Function]}
      >
         
        <span>
          Se test-sang
        </span>
      </a>
    </div>
  </nav>
  <div
    className="css-9c3fle"
  >
    <div
      className="css-cssveg"
    >
      <div
        className="css-0"
      >
        <form
          action="/search"
          className="css-ofpocj"
          onChange={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="css-1rmzhqg"
          >
            <div
              className="css-1f7apd6"
            >
              <div
                className="css-18a5p6x"
                focused={0}
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </div>
              <input
                autoComplete="off"
                className="css-psh2zg"
                name="query"
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                open={0}
                placeholder="Søk på sang"
                type="search"
                value=""
              />
            </div>
          </div>
        </form>
      </div>
    </div>
    <main
      className="css-1tbz3yy"
    >
      <div
        className="css-2x3ewd"
      >
        <div
          className="css-1nqoz7o"
        >
          <div
            className="css-cssveg"
          >
            <div
              className="css-cssveg"
            >
              <div
                className="css-c8mo9m"
                onClick={[Function]}
              />
              <div
                className="MuiTabs-root css-bgtre9-MuiTabs-root"
                showindicator={1}
              >
                <div
                  className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
                  onScroll={[Function]}
                  style={
                    Object {
                      "marginBottom": -0,
                      "overflow": "hidden",
                    }
                  }
                >
                  <div
                    aria-label="filter tabs"
                    aria-orientation={null}
                    className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                    onKeyDown={[Function]}
                    role="tablist"
                  >
                    <button
                      aria-controls="filter-tabpanel-0"
                      aria-selected={false}
                      className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-rflsc5-MuiButtonBase-root-MuiTab-root"
                      disabled={false}
                      id="filter-tab-0"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="tab"
                      style={
                        Object {
                          "marginLeft": "1rem",
                        }
                      }
                      tabIndex={-1}
                      type="button"
                    >
                      
                    </button>
                    <button
                      aria-controls="filter-tabpanel-1"
                      aria-selected={true}
                      className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-rflsc5-MuiButtonBase-root-MuiTab-root"
                      disabled={false}
                      id="filter-tab-1"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="tab"
                      tabIndex={0}
                      type="button"
                    >
                      Tema
                    </button>
                  </div>
                  <span
                    className="MuiTabs-indicator css-yihvlh-MuiTabs-indicator"
                    style={
                      Object {
                        "left": 0,
                        "width": 0,
                      }
                    }
                  />
                </div>
              </div>
              <div
                aria-labelledby="filter-tab-1"
                hidden={false}
                id="filter-tabpanel-1"
                role="tabpanel"
              >
                <ul
                  className=" css-1bjoovq"
                  horizontalScroll={false}
                />
              </div>
            </div>
            <div
              className="css-1lydpzv"
            >
              <div
                className="MuiFormControl-root MuiFormControl-fullWidth css-15zqp5e-MuiFormControl-root"
              >
                <div
                  className="css-1y76fht"
                >
                  <div
                    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-tjat9u-MuiInputBase-root-MuiOutlinedInput-root"
                    onClick={[Function]}
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-labelledby="sort-select-label sort-select"
                      className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-1jz9ynx-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                      id="sort-select"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onMouseDown={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      Sorter på 
                      nyeste
                    </div>
                    <input
                      aria-hidden={true}
                      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                      disabled={false}
                      name="sort"
                      onAnimationStart={[Function]}
                      onChange={[Function]}
                      required={false}
                      tabIndex={-1}
                      value="releaseDate--desc"
                    />
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1wu7ecg-MuiSvgIcon-root-MuiSelect-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                    <fieldset
                      aria-hidden={true}
                      className="MuiOutlinedInput-notchedOutline css-9425fu-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        className="css-1a1ihhj"
                      >
                        <span
                          className="notranslate"
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "&#8203;",
                            }
                          }
                        />
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <button
                className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root css-9fezxy-MuiButtonBase-root-MuiButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                Nullstill
                <span
                  className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <div
              style={
                Object {
                  "opacity": 0,
                }
              }
            >
              <div
                className="css-inmhno"
              >
                <span
                  className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1arrlox-MuiCircularProgress-root"
                  role="progressbar"
                  style={
                    Object {
                      "height": 40,
                      "width": 40,
                    }
                  }
                >
                  <svg
                    className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                      cx={44}
                      cy={44}
                      fill="none"
                      r={20.2}
                      strokeWidth={3.6}
                      style={Object {}}
                    />
                  </svg>
                </span>
              </div>
            </div>
            <div
              style={
                Object {
                  "opacity": 0,
                }
              }
            >
              <div />
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`rendered successfully with query 1`] = `
<div
  className="css-185wnrw"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <nav
    className="css-1ebp07k"
    open={false}
  >
    <div
      className="css-ydsbw5"
    >
      <a
        className="css-1yg7sek"
        href="/"
        onClick={[Function]}
      >
         
        <span>
          Hjem
        </span>
      </a>
      <a
        className="css-1yg7sek"
        href="/search"
        onClick={[Function]}
      >
         
        <span>
          Sanger
        </span>
      </a>
      <a
        className="css-1yg7sek"
        href="/search?query=søker på dette"
        onClick={[Function]}
      >
         
        <span>
          Søk på noe
        </span>
      </a>
      <a
        className="css-1yg7sek"
        href="/song/testsang"
        onClick={[Function]}
      >
         
        <span>
          Se test-sang
        </span>
      </a>
    </div>
  </nav>
  <div
    className="css-9c3fle"
  >
    <div
      className="css-cssveg"
    >
      <div
        className="css-0"
      >
        <form
          action="/search"
          className="css-ofpocj"
          onChange={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="css-1rmzhqg"
          >
            <div
              className="css-1f7apd6"
            >
              <div
                className="css-18a5p6x"
                focused={0}
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </div>
              <input
                autoComplete="off"
                className="css-psh2zg"
                name="query"
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                open={0}
                placeholder="Søk på sang"
                type="search"
                value="all pris"
              />
            </div>
          </div>
        </form>
      </div>
    </div>
    <main
      className="css-1tbz3yy"
    >
      <div
        className="css-2x3ewd"
      />
    </main>
  </div>
</div>
`;

exports[`rendered successfully without results 1`] = `
<div
  className="css-185wnrw"
  style={
    Object {
      "height": "768px",
    }
  }
>
  <nav
    className="css-1ebp07k"
    open={false}
  >
    <div
      className="css-ydsbw5"
    >
      <a
        className="css-1yg7sek"
        href="/"
        onClick={[Function]}
      >
         
        <span>
          Hjem
        </span>
      </a>
      <a
        className="css-1yg7sek"
        href="/search"
        onClick={[Function]}
      >
         
        <span>
          Sanger
        </span>
      </a>
      <a
        className="css-1yg7sek"
        href="/search?query=søker på dette"
        onClick={[Function]}
      >
         
        <span>
          Søk på noe
        </span>
      </a>
      <a
        className="css-1yg7sek"
        href="/song/testsang"
        onClick={[Function]}
      >
         
        <span>
          Se test-sang
        </span>
      </a>
    </div>
  </nav>
  <div
    className="css-9c3fle"
  >
    <div
      className="css-cssveg"
    >
      <div
        className="css-0"
      >
        <form
          action="/search"
          className="css-ofpocj"
          onChange={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="css-1rmzhqg"
          >
            <div
              className="css-1f7apd6"
            >
              <div
                className="css-18a5p6x"
                focused={0}
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </div>
              <input
                autoComplete="off"
                className="css-psh2zg"
                name="query"
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                open={0}
                placeholder="Søk på sang"
                type="search"
                value="songnotfound"
              />
            </div>
          </div>
        </form>
      </div>
    </div>
    <main
      className="css-1tbz3yy"
    >
      <div
        className="css-2x3ewd"
      />
    </main>
  </div>
</div>
`;
